src/posix/sdl/hardware.cpp

```diff
@@ -334,8 +334,8 @@ CUSTOM_CVAR(Bool, swtruecolor, true, CVAR_ARCHIVE|CVAR_GLOBALCONFIG|CVAR_NOINITC
 
 CUSTOM_CVAR (Bool, fullscreen, false, CVAR_ARCHIVE|CVAR_GLOBALCONFIG)
 {
-	NewWidth = screen->VideoWidth;
-	NewHeight = screen->VideoHeight;
+	NewWidth = screen->GetWidth();
```

Comment:
It comes from a nasty bug that I just committed a fix for, which was also happening on Linux.

See commit https://github.com/coelckers/gzdoom/commit/d8c6966917ad207276a402aefec847703ec8c26e